src/errors/errors.ts

```diff
@@ -0,0 +1,215 @@
+import type { Class } from '@matrixai/errors';
+import type { JSONValue } from '@/types';
+import { AbstractError } from '@matrixai/errors';
+
+interface RPCError extends Error {}
+
+class ErrorRPC<T> extends Error implements RPCError {
```

Comment:
replace errorCode with just code,
create an enum to reference the standard recommended codes from the spec:
https://www.jsonrpc.org/specification#error_object